bottle.py

```diff
@@ -2425,7 +2425,7 @@ def load_config(self, filename, **options):
         for section in conf.sections():
             for key in conf.options(section):
                 value = conf.get(section, key)
-                if section not in ['bottle', 'ROOT']:
+                if section not in ('bottle', 'ROOT'):
```

Comment:
@RonRothman, just run this
```
from dis import dis

def f():
    return [1, 2, 3]

def g():
    return (1, 2, 3)

dis(f)
print('-' * 60)
dis(g)
```

Reply:
@RonRothman but in this case it no matter on my python 3.8.5 (I just checkd)!
```
def f(section):
    return section not in ['bottle', 'ROOT']

def g(section):
    return section not in ('bottle', 'ROOT')
```

Reply:
What difference did you expect?

Reply:
Did you run this samples? Anyway, for the 1st case:
```
  4           0 LOAD_CONST               1 (1)
              2 LOAD_CONST               2 (2)
              4 LOAD_CONST               3 (3)
              6 BUILD_LIST               3
              8 RETURN_VALUE
------------------------------------------------------------
  7           0 LOAD_CONST               1 ((1, 2, 3))
              2 RETURN_VALUE
```
Difference is obvious. But in 2nd case no any `BUILD_LIST`:
```
  4           0 LOAD_FAST                0 (section)
              2 LOAD_CONST               1 (('bottle', 'ROOT'))
              4 COMPARE_OP               7 (not in)
              6 RETURN_VALUE
------------------------------------------------------------
  7           0 LOAD_FAST                0 (section)
              2 LOAD_CONST               1 (('bottle', 'ROOT'))
              4 COMPARE_OP               7 (not in)
              6 RETURN_VALUE
```

Reply:
What I'm wondering is: what difference does it make? Are you hoping to save 2 nanoseconds of runtime?

Reply:
Interesting is that "BUILD_LIST" is also in the tuple case if you change the order of `dis` calls:

```
 $ python
Python 3.6.12 (default, Nov 28 2020, 20:38:55) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from dis import dis
>>> 
>>> def g():
...     return (1, 2, 3)
... 
>>> def f():
...     return [1, 2, 3]
... 
>>> 
>>> dis(f)
  2           0 LOAD_CONST               1 (1)
              2 LOAD_CONST               2 (2)
              4 LOAD_CONST               3 (3)
              6 BUILD_LIST               3
              8 RETURN_VALUE
>>> print('-' * 60)
------------------------------------------------------------
>>> dis(g)
  2           0 LOAD_CONST               4 ((1, 2, 3))
              2 RETURN_VALUE
>>> 
$ python
Python 3.6.12 (default, Nov 28 2020, 20:38:55) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from dis import dis
>>> 
>>> def f():
...     return [1, 2, 3]
... 
>>> def g():
...     return (1, 2, 3)
... 
>>> dis(f)
  2           0 LOAD_CONST               1 (1)
              2 LOAD_CONST               2 (2)
              4 LOAD_CONST               3 (3)
              6 BUILD_LIST               3
              8 RETURN_VALUE
>>> print('-' * 60)
------------------------------------------------------------
>>> dis(g)
  2           0 LOAD_CONST               4 ((1, 2, 3))
              2 RETURN_VALUE
>>> 
```
Nevertheless, `tuple`s are slightly faster to create, because they use 1 block of memory compared to lists. Also, the need a bit less memory.  For small lists it's really negligible, but for a website handling many requests this can add up very quickly.

https://medium.com/@prajeen.v/why-tuples-are-faster-and-when-to-use-list-f2f07291e923

Reply:
@RonRothman, you right - no difference, it's my own obsession.
Like cpyton's developers obsession - see bytecode above. 

Anyway, you can propose me more ambitious goal - to save 2 milliseconds, or 2 seconds - if you show me bottleneck )